File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/postgresql/DomainType.scala
package dev.typr.foundations.docs.postgresql
import dev.typr.scalafoundations.*
import dev.typr.scalafoundations.data.*



@SuppressWarnings(Array("unused"))
object DomainType:
  //start
  // Wrapper type
  case class Email(value: String)

  // Create PgType from text
  val emailType: PgType[Email] = PgTypes.text.bimap(Email.apply, _.value)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/postgresql/DomainType.java
package dev.typr.foundations.docs.postgresql;

import dev.typr.foundations.PgType;
import dev.typr.foundations.PgTypes;

@SuppressWarnings("unused")
public class DomainType {
    //start
    // Wrapper type
    public record Email(String value) {}

    // Create PgType from text
    PgType<Email> emailType = PgTypes.text.transform(Email::new, Email::value);
    //stop
}
